### PR TITLE
Refactor eeprom handling

### PR DESCRIPTION
**Pull request by elcojacobs**
tracking, do not merge yet

### CODE REVIEW

File: lib/cbox/inc/cbox/Object.hpp [hunk 1]
```diff
@@ -63,10 +63,10 @@ class Object {
 public:
     Object() = default;
```
Comment (steersbob) [inline lib/cbox/inc/cbox/Object.hpp:64]: May as well declare none, if they're all default.

File: lib/cbox/inc/cbox/Serialization.hpp [hunk 2]
```diff
@@ -51,33 +51,40 @@ class Request {
...
-{
-    if (pos + sizeof(T) > out.size()) {
-        return false;
-    }
-    auto p = reinterpret_cast<const uint8_t*>(std::addressof(v));
-    std::memcpy(out.data() + pos, p, sizeof(T));
     return true;
 }
 
+// template <typename T>
```
Comment (steersbob) [inline lib/cbox/inc/cbox/Serialization.hpp:66]: Remove commented code

File: lib/cbox/src/EepromDataStream.cpp [hunk 3]
```diff
@@ -18,62 +18,60 @@
  * along with Brewblox. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "cbox/EepromDataStream.hpp"
-#include "cbox/Crc.hpp"
-#include <algorithm>
+// #include "cbox/EepromDataStream.hpp"
```
Comment (steersbob) [inline lib/cbox/src/EepromDataStream.cpp:21]: Looks like the file can be removed

File: test/cbox/makefile [hunk 4]
```diff
@@ -58,8 +58,11 @@ CFLAGS += -ffunction-sections -Wall
 # Flag compiler error for [-Wdeprecated-declarations]
 CFLAGS += -Werror=deprecated-declarations
 
+# stop on first error
+CFLAGS += -Wfatal-errors
+
 # Generate dependency files automatically.
-CFLAGS += -g -Og
+CFLAGS += -g -O0
```
Comment (steersbob) [inline test/cbox/makefile:65]: Why?

**Review by steersbob — APPROVED** [review]
Cleanup mostly. Looks good otherwise.

File: test/cbox/makefile (hunk 4 above)
Comment (elcojacobs) [reply test/cbox/makefile:65]: Debugging, can be reverted

File: lib/cbox/inc/cbox/Object.hpp (hunk 1 above)
Comment (elcojacobs) [reply lib/cbox/inc/cbox/Object.hpp:64]: instructor is declared virtual, which is not default

File: test/cbox/makefile (hunk 4 above)
Comment (steersbob) [reply test/cbox/makefile]: `-Og` is the debug-friendly variant.

**Pull request merged**